src/include/brain/util/model_util.h

```diff
@@ -0,0 +1,97 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// model_util.h
+//
+// Identification: src/include/brain/util/model_util.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "brain/util/eigen_util.h"
+
+namespace peloton {
+namespace brain {
+
+// Forward declarations
+class BaseForecastModel;
+
+class ModelUtil {
+ public:
+  /**
+   * Simple MSE loss function: mean((y - pred)**2)
+   * @param ytrue: True labels
+   * @param ypred: Predicted labels
+   * @return: loss value
+   */
+  static float MeanSqError(const matrix_eig &ytrue, const matrix_eig &ypred);
+
+  /**
+   * Function to check for early stopping criteria. This API is similar to
+   * **keras.callbacks.EarlyStopping** and based on below references
+   * @param val_losses: vector of validation losses
+   * @param patience: number of epochs to wait for a threshold change in loss
+   * @param delta: minimum needed threshold change to continue training
+   * @return: True if one should stop.
+   * Refences: (1) https://stats.stackexchange.com/q/231061/72908
+   *           (2) Check: "Early Stopping - but when?" for a more detailed
+   *           suite of methods.
+   */
+  static bool EarlyStop(vector_t val_losses, size_t patience = 10, float delta = 1e-2);
```

Comment:
Eh this is not straightforward and easy to confuse people. Please put these default numbers in a central parameter place. For example `workload_defaults.h ` is fine.